kinder/pkg/cluster/manager/actions/kubeadm-init.go

```diff
@@ -101,9 +101,6 @@ func kubeadmInit(cp1 *status.Node, copyCertsMode CopyCertsMode, patchesDir, igno
 			// NB. certificate key is passed via the config file)
 		)
 	}
-	if patchesDir != "" {
-		initArgs = append(initArgs, "--experimental-patches", constants.PatchesDir)
-	}
```

Comment:
for a period of time we will have tests for 1.20, 1.21, 1.22, 1.23

this means:
- 1.20, 1.21, must use --experimental-patches
- "init"/"join" for 1.22, 1.23 can use v1beta3 / patches from config only.
- "upgrade" for 1.22, 1.23 must use --patches

so it's a bit trickier.

see for example on how to branch on versions:
https://github.com/kubernetes/kubeadm/blob/a29017a17040d7dd2f10eac92647e9aaed26c300/kinder/pkg/cluster/manager/actions/kubeadm-init.go#L44-L51

Reply:
I updated the PR to check the version before applying --patch.